yogo/packages/core/src/app.e2e.test.ts

```diff
@@ -0,0 +1,125 @@
+import { bootstrap } from "./bootstrap";
+import { ApolloServer } from "apollo-server";
+import {
+  createTestClient,
+  ApolloServerTestClient
+} from "apollo-server-testing";
+
+describe("e2e", () => {
+  let server: ApolloServer;
+  let url: string;
+  let client: ApolloServerTestClient | null = null;
+
+  beforeAll(async () => {
+    console.log("bootstrapping a server");
+    ({ server, url } = await bootstrap());
+    client = createTestClient(server);
+  });
+  afterAll(async () => {
+    console.log("shutting down the server");
+    await server.stop();
+    client = null;
+  });
+  it("server is defined", () => {
+    expect(server).toBeDefined();
+    expect(url).toBe("http://localhost:4000/");
+  });
+  it("client is defined", () => {
+    expect(client).toBeDefined();
+  });
+
+  describe("Group", () => {
+    it("creates groups", async () => {
+      const res = await client?.query({
+        query: `
+        mutation {
+          createGroup(id: "my-group"){
+            id
+            notes{
+              id
+              freq
+              duration
+              offset
+            }
+          }
+        }
+        `
+      });
+      expect(res?.data).toMatchInlineSnapshot(`
+        Object {
+          "createGroup": Object {
+            "id": "my-group",
+            "notes": Array [],
+          },
+        }
+      `);
+    });
+    it("finds groups", async () => {
+      const res = await client?.query({
+        query: `
+          query{
+            findGroupBy(id: "my-group"){
+              id
+              notes {
+                id
+                freq
+                duration
+                offset
+              }
+            }
+          }
+        `
+      });
+      expect(res?.data).toMatchInlineSnapshot(`
+Object {
+  "findGroupBy": Object {
+    "id": "my-group",
+    "notes": Array [],
+  },
+}
+`);
+    });
+    it("pushes notes to groups", async () => {
+      const res = await client?.mutate({
+        mutation: `
+        mutation {
+          pushNote(id:"my-group", payload: {
```

Comment:
This is where I ponder:

The note creation feature could be desired in two major situations.

A. Create a note, then link it with (a) group(s).
B. Create notes **directly** on (a) group(s).

In the case of A, a client can **group-agnostically** generate notes. How to attach notes to which groups is totally deligated to the later process. The downside of this approach is the inevitability of the increasing number of the API round-trips.
The case B solves exactly this, by generating notes directly on a pre-existing group. The downside of this approach is that the client must know/track the name of the group. (Or, we could find-or-create the group entity to solve this)

Anyways, both situations seem pretty common. 
Luckily, GraphQL is very flexible in expanding API, so I'm thinking that we could support both.

cc: @zigen @szk2s